llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -592,8 +663,223 @@ static Error applyAddends(MCCASReader &Reader,
+      break;
+    }
+    case dwarf::DW_FORM_strx3: {
+      FormSize += 3;
+      break;
+    }
+    case dwarf::DW_FORM_data4:
+    case dwarf::DW_FORM_ref4:
+    case dwarf::DW_FORM_ref_sup4:
+    case dwarf::DW_FORM_strx4:
+    case dwarf::DW_FORM_addrx4: {
+      FormSize += 4;
+      break;
+    }
+    case dwarf::DW_FORM_data8:
+    case dwarf::DW_FORM_ref8:
+    case dwarf::DW_FORM_ref_sup8: {
+      FormSize += 8;
+      break;
+    }
+    case dwarf::DW_FORM_data16: {
+      FormSize += 16;
+      break;
+    }
+    case dwarf::DW_FORM_sdata: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      DWARFExtractor.getSLEB128(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_udata:
+    case dwarf::DW_FORM_ref_udata:
+    case dwarf::DW_FORM_rnglistx:
+    case dwarf::DW_FORM_loclistx:
+    case dwarf::DW_FORM_GNU_addr_index:
+    case dwarf::DW_FORM_GNU_str_index:
+    case dwarf::DW_FORM_addrx:
+    case dwarf::DW_FORM_strx: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      DWARFExtractor.getULEB128(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_LLVM_addrx_offset: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      DWARFExtractor.getULEB128(&CUOffset);
+      FormSize += CUOffset - PrevOffset + 4;
+      break;
+    }
+    case dwarf::DW_FORM_string: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      auto CurrOffset = CUOffset;
+      DWARFExtractor.getCStr(&CUOffset);
+      FormSize += CUOffset - CurrOffset;
+      break;
+    }
+    case dwarf::DW_FORM_indirect: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      FormVal =
+          static_cast<dwarf::Form>(DWARFExtractor.getULEB128(&CUOffset, &Err));
+      Indirect = true;
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_strp:
+    case dwarf::DW_FORM_sec_offset:
+    case dwarf::DW_FORM_GNU_ref_alt:
+    case dwarf::DW_FORM_GNU_strp_alt:
+    case dwarf::DW_FORM_line_strp:
+    case dwarf::DW_FORM_strp_sup: {
+      FormSize += FP.getDwarfOffsetByteSize();
+      break;
+    }
+    case dwarf::DW_FORM_ref_sig8: {
+      FormSize += 8;
+      break;
+    }
+    case dwarf::DW_FORM_implicit_const:
+    case dwarf::DW_FORM_flag_present: {
+      FormSize += 0;
+      break;
+    }
+    default:
+      llvm_unreachable("usupported form");
```

Comment:
we should avoid `default`s, otherwise we don't get compiler warnings when new entries are added to the enumeration. We're only missing two forms from this list, so we can just put them here:

`DW_FORM_addrx3` and `DW_FORM_lo_user`